src/twitter.d.ts

```diff
@@ -0,0 +1,64 @@
+declare module 'twitter-v2' {
+  export interface ApplicationCredentials {
+    consumer_key: string;
+    consumer_secret: string;
+    bearer_token?: string;
+  }
+
+  export interface UserCredentials {
+    consumer_key: string;
+    consumer_secret: string;
+    access_token_key: string;
+    access_token_secret: string;
+  }
+
+  export type CredentialsArgs = ApplicationCredentials | UserCredentials;
+
+  export interface RequestParameters {
+    [key: string]: any;
+  }
+
+  export interface TwitterResponse<T extends any> {
+    data: T;
+  }
+
+  export class Credentials implements ApplicationCredentials, UserCredentials {
```

Comment:
Organization-wise let's define this and the associated interfaces in `Credentials.d.ts`